src/blog/2023/06/Flowforge-as-a-No-Code-Ethernet_IP-to-Profinet-Protocol-Converter.md

```diff
@@ -0,0 +1,208 @@
+
+Populate the connection properties to match your hardware. The cycle time is updated to 1000ms to match the cycle time of our `eth-ip in` nodes. You can adjust this value to match your intended application.
+
+![18](./images/ethip-to-profinet/e-to-p-18.png)
+
+On the `Variables` tab, some special formatting is required to point to the absolute reference of the tag DB location in the S7 PLC.
+
+For information on how to format S7 absolute tag references in a way the `s7 endpoint` node is expecting, refer to the node documentation for further information.
+
+[https://flows.nodered.org/node/node-red-contrib-s7](https://flows.nodered.org/node/node-red-contrib-s7)
+
+For reference, here is an example of how we set the tags in our stacklight PLC example and how it looks in our `s7 endpoint`.
+
+![19](./images/ethip-to-profinet/e-to-p-19.png)
+
+Once the tags are populated we can select our configured endpoint from the dropdown list, point to our first variable, `Conveyor_RTS`, and give the node a name.
+
+![20](./images/ethip-to-profinet/e-to-p-20.png)
+
+Repeat this process for the remaining tags.
+
+![21](./images/ethip-to-profinet/e-to-p-21.png)
+
+# Test the Conversion
+
+The only thing remaining is to simply wire the nodes together, and confirm the values pass through.
+
+![22](./images/ethip-to-profinet/e-to-p-22.png)
+
+Manipulate the incoming values and confirm the data passes through as expected. Because of the report by exception nature of the `eth-ip in` node, tag changes should be near instantaneous on the receiving PLC.
+
+We can stop here, but we can improve this flow by adding a `filter` node on our REAL data-type, `Robot_Position`.
+
+## Add Filter to REAL data
+
+Depending on how noisy the REAL data is (common with unfiltered 4-20mA field transmitters), and how much granularity you need to capture, it is good practice to add a filter on REAL data to reduce fieldbus traffic coming out of our soft protocol converter.
+
+![23](./images/ethip-to-profinet/e-to-p-23.png)
+
+In the example above, we arbitrarily applied a 3% deadband to the `Robot_Position` value, which means that the value must change by greater than or equal to 3% compared to the last input value, or else the data will be discarded before being sent to the stacklight PLC.
+
+You can adjust the deadband to find the right balance for your particular application.
+
+We can see the effect the deadband filter had by adding debug nodes before and after the filter.
+
+![24](./images/ethip-to-profinet/e-to-p-24.png)
+
+As shown above, when `Robot_Position` changed from 15.6 to 15.6999..., the value was captured on the input of the filter, but was discarded on the output.
+
+When the `Robot_Position` went from 15.6999 to 18, the filter allowed it to pass as it exceeded the deadband limit we had set.
+
+Use filters to optimize your fieldbus converter network performance, especially if dealing with noisy signals or large quantities of REAL datatypes.
```

Comment:
I think a conclusion would be great, summerize what was learned etc. Do you want to post the JSON of the flow?

Reply:
- accepted fixes to grammar/capitalization errors
- added descriptions to all pictures
- added conclusion section
- added JSON source code at end of article
- added art request -https://github.com/flowforge/website/issues/872